### PR TITLE
Initial src code

### PR DESCRIPTION
**Pull request by weikinhuang**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,9 +4,10 @@
   "description": "Jasmine test helper for detecting setTimeout and setInterval usage outside of test boundaries",
   "main": "dist/jasmine-stray-timers.js",
   "scripts": {
-    "test": "npm run eslint && npm run karma -- --single-run",
+    "test": "npm run eslint && npm run karma -- --single-run --reporters=progress,coverage",
```
Comment (kaicataldo) [inline package.json:7]: Is the extra `--` intentional here?

File: package.json [hunk 2]
```diff
@@ -4,9 +4,10 @@
...
+    "test:dev": "npm run karma -- --reporters=mocha,coverage",
```
Comment (kaicataldo) [inline package.json:8]: Same as above

File: src/timers.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+const realClearInterval = window.clearInterval;
+
+// export real timers for testing
+export { realSetTimeout, realSetInterval };
+
+var runningTimeouts = [];
+var runningIntervals = [];
+
+/**
+ * Mark an timer as executed inside the test constraints
```
Comment (kaicataldo) [inline src/timers.js]: `Mark an timer` -> `Mark a timer`

File: test/jasmine/jasmineClock.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  beforeEach(function() {
+    jasmine.clock().install();
+  });
+
+  afterEach(function() {
+    jasmine.clock().uninstall();
+  });
+
+  it('#overrideJasmineClock should work as the without throws', function(done) {
```
Comment (kaicataldo) [inline test/jasmine/jasmineClock.js]: Typo? Not sure what this means 😄

File: package.json [hunk 5]
```diff
@@ -4,9 +4,10 @@
...
     "eslint": "eslint src/**/*.js test/**/*.js",
-    "karma": "karma start test/karma.conf.js"
+    "karma": "cross-env NODE_ENV=test karma start"
```
Comment (designjockey) [inline package.json:10]: TIL cross-env, pretty cool

File: src/timers.js (hunk 3 above)
Comment (designjockey) [inline src/timers.js]: Mark **a** timer

File: package.json (hunk 1 above)
Comment (weikinhuang) [reply package.json:7]: yes, prevents npm arguments by accident

File: test/jasmine/jasmineClock.js (hunk 4 above)
Comment (weikinhuang) [reply test/jasmine/jasmineClock.js]: i mean `as intended`

File: src/timers.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+  return realClearTimeout.apply(this, arguments);
+}
+
+/**
+ * Mark an interval as cleared inside the test constraints
+ *
+ * @param {Number} timerId
+ */
+function removeInterval(timerId) {
+  runningIntervals = runningIntervals.filter(({ id }) => id !== timerId);
```
Comment (designjockey) [inline src/timers.js:71]: `.filter(({ id }) => id !== timerId)` same is used on line 18, consider extracting to a function.

**Review by kaicataldo — APPROVED** [review]
LGTM

File: test/jasmine/jasmineClock.js (hunk 4 above)
Comment (designjockey) [inline test/jasmine/jasmineClock.js]: lol, not sure what `should work as the without throws` means, typo?

File: test/jasmine/jasmineClock.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    expect(spy).not.toHaveBeenCalled();
+    jasmine.clock().tick(16);
+    expect(spy).toHaveBeenCalled();
+
+    // to make sure we don't throw in the test
+    realSetTimeout(done, 25);
+  });
+
+  it('#overrideJasmineClock should work as the without throws', function(done) {
```
Comment (designjockey) [inline test/jasmine/jasmineClock.js]: Same, `should work as the without throws`

File: src/timers.js (hunk 6 above)
Comment (weikinhuang) [reply src/timers.js:71]: It uses a local var

**Review by designjockey — APPROVED** [review]

**Pull request merged**